src/settings/MappingProfiles/MappingProfilesForm.js

```diff
@@ -11,55 +11,67 @@ import {
   intlShape,
 } from 'react-intl';
 
-import {
-  Field,
-  change,
-} from 'redux-form';
+import { connect } from 'react-redux';
+import { change } from 'redux-form';
 import {
   get,
   identity,
+  isEmpty,
 } from 'lodash';
 
-import {
-  Select,
-  Headline,
-  TextArea,
-  TextField,
-  Accordion,
-  AccordionSet,
-} from '@folio/stripes/components';
 import stripesForm from '@folio/stripes/form';
 
 import {
-  validateRequiredField,
-  compose, withProfileWrapper,
+  compose,
+  usePrevious,
+  withProfileWrapper,
 } from '../../utils';
 import {
   ENTITY_KEYS,
   LAYER_TYPES,
   PROFILE_TYPES,
 } from '../../utils/constants';
 import {
-  FullScreenForm,
-  FolioRecordTypeSelect,
-  ProfileAssociator,
   INCOMING_RECORD_TYPES,
-  FOLIO_RECORD_TYPES,
+  FOLIO_RECORD_TYPES, FlexibleForm,
 } from '../../components';
 
+import {
+  getInitialDetails,
+  getInitialFields,
+  getReferenceTables,
+} from './initialDetails';
+
+// @TODO: Remove this after server-side configs will be available
+import { formConfigSamples } from '../../../test/bigtest/mocks';
+
+import './MappingProfiles.css';
+
 const formName = 'mappingProfilesForm';
 
 export const MappingProfilesFormComponent = ({
   intl: { formatMessage },
   pristine,
   submitting,
   initialValues,
+  mappingDetails,
   location: { search },
   handleSubmit,
   onCancel,
   dispatch,
 }) => {
   const { profile } = initialValues;
+  const { layer } = queryString.parse(search);
+
+  const isEditMode = layer === LAYER_TYPES.EDIT;
+  const isSubmitDisabled = pristine || submitting;
+  const formConfig = formConfigSamples.find(cfg => cfg.name === formName);
+  const associations = [
+    ...[],
```

Comment:
I think this is excess. Please clarify it is need if you think otherwise.

Reply:
Done